cachito/web/utils.py

```diff
@@ -1,7 +1,33 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
+from collections import OrderedDict
+
 from flask import request, url_for
 
 
+def deep_sort(orig_item):
+    """
+    Return a new element recursively sorted in ascending order.
+
+    The function sorts recursively only dicts & lists,
+    anything that is not a list or a dict will be returned unchanged.
+
+    :param orig_item: Original dict or list
+    :return: Recursively sorted dict or list according to orig_item
+    :rtype: Any
+    """
+    if isinstance(orig_item, list):
+        sorted_item = [deep_sort(item) for item in orig_item]
+        if len(sorted_item) and isinstance(sorted_item[0], OrderedDict):
+            sorted_item = sorted(sorted_item, key=lambda ordered_dict: ordered_dict.items())
+        else:
+            sorted_item = sorted(sorted_item)
```

Comment:
We assume here that lists will always contain only one type of element, otherwise we would not be able to compare the elements. This seems like a reasonable assumption, but it makes me wonder if we even need to sort lists. @athos-ribeiro what exactly do we need to sort?

Reply:
It is reasonable to sort the lists as well so we don't need to rely on the implementation for fetching a package dependencies from the DB.

in this sense, we'd sort the following structure

```
{
  "metadata": DICT_WITH_STRING_AND_INT_ENTRIES,
  "image_contents": [
    {
    "purl": STRING,
    "dependencies": [ {"purl": STRING}, {"purl": STRING} ],
    "sources": [{"purl": STRING}, {"purl": STRING}]
    }
 ]
}
```

Reply:
@akhmanova, it would be nice to have a test dedicated to this feature, with additional entries in the image_contents array.

Reply:
@athos-ribeiro @chmeliik So we do not need this part? 
```
else:
            sorted_item = sorted(sorted_item)
```
we can raise an exception in case it is not an `OrderedDict` in `list`, because I can not see any lists except:
```
[
    {},
    {},
    ...
]
```

Reply:
@akhmanova Yeah I guess we don't, though in that case it might be a good idea to rename this function to `deep_sort_icm` (or just `sort_icm`) and update the description to say that it is meant for sorting ICMs

Reply:
Ok, will do. I found a mistake, I am fixing and testing it right now.